### ISSUE TITLE
[Feature] Dynamic Pricing Assistant - AI-Powered Competitive Price Optimization

### ISSUE THREAD

**Issue by Techhackontime999** [Techhackontime999/Shop-Seed-Art#4]
# Feature: Dynamic Pricing Assistant

## Overview
Machine learning algorithm that analyzes uploaded product image and description to suggest an optimal, competitive selling price based on current market trends, raw material costs, and comparable listings.

## User Problem
Artisans struggle with:
- Knowing fair market price for their unique handmade products
- Understanding competitor pricing on marketplaces
- Factoring in raw material cost fluctuations
- Balancing profitability with competitiveness
- Dynamic pricing for seasons, festivals, demand spikes

## Solution
AI pricing engine that:
1. **Analyzes product image** (visual features: material, complexity, craftsmanship quality)
2. **Processes description** (extracted attributes: material, dimensions, technique, origin)
3. **Queries market data** (comparable listings, historical sales, trend analysis)
4. **Calculates cost baseline** (raw material + labor + overhead estimates)
5. **Suggests optimal price** with confidence score and reasoning
6. **Provides price range** (min/floor, recommended, max/premium)
7. **Monitors & alerts** on price optimization opportunities

### Backend (Django - new `ai_pricing` app)
- **Models**:
  - `PricingJob` - track pricing analysis requests
  - `MarketPriceSnapshot` - cached competitor pricing data
  - `RawMaterialCostIndex` - tracked material costs (cotton, silk, wool, clay, metals, etc.)
  - `PricingModelVersion` - model versioning for A/B testing
  - `SellerPricingFeedback` - accepted/rejected suggestions for retraining
- **Services**:
  - `VisualAnalysisService` - CNN for material/complexity/quality scoring from images
  - `MarketDataService` - scrape/cache marketplace prices (GeM, Amazon, Flipkart, Etsy, niche)
  - `CostEstimationService` - material + labor cost models per category/region
  - `PricingInferenceService` - main ML model (XGBoost/LightGBM + neural net ensemble)
  - `PriceMonitoringService` - scheduled re-evaluation, alert on opportunities
- **API Endpoints**:
  - `POST /api/ai/pricing/analyze/` - submit product (image + description) → price suggestion
  - `GET /api/ai/pricing/history/{product_id}/` - price recommendation history
  - `POST /api/ai/pricing/feedback/` - seller accepts/rejects/modifies suggestion
  - `GET /api/ai/pricing/material-costs/{category}/` - current material cost index

### ML Pipeline
```
Input: Product Image(s) + Description (structured attributes)
    ↓
Visual Encoder (ResNet/EfficientNet → embedding: material, texture, complexity, quality)
Text Encoder (BERT/IndicBERT → embedding: attributes, story, uniqueness)
    ↓
Feature Fusion + Market Context (competitor prices, seasonality, demand signals)
    ↓
Cost Baseline Calculator (material index × quantity + labor hours × regional wage)
    ↓
Pricing Model (Gradient Boosting + Neural Net ensemble)
    ↓
Output: {
    "floor_price": 450,           // Below this = loss
    "recommended_price": 799,     // Optimal conversion × margin
    "premium_price": 1199,        // For premium positioning
    "confidence": 0.87,
    "reasoning": [
        "Similar handloom cotton sarees sell at 650-950",
        "Raw cotton cost up 12% this quarter",
        "Festival season demand +15% typical",
        "Your craftsmanship quality scores 0.82/1.0"
    ],
    "comparables": [...],
    "cost_breakdown": {"material": 280, "labor": 120, "overhead": 50}
}
```

### Data Sources
- **Marketplace APIs**: GeM, Amazon PA-API, Flipkart Affiliate, Etsy API
- **Web Scraping**: Niche handicraft sites, exhibition catalogs
- **Government Data**: Raw material price indices (cotton, silk, jute, wool)
- **Platform Data**: Own historical sales, conversion rates, seller feedback
- **Seasonal Calendar**: Festival calendar, wedding seasons, export seasons

### Pricing Strategies per Category
| Category | Strategy |
|----------|----------|
| Textiles | Per-meter + design complexity + brand premium |
| Handicrafts | Time-based (hours) + material + uniqueness factor |
| Jewelry | Material weight × rate + making charges + design premium |
| Pottery | Size + firing complexity + glaze + food-safe certification |

### Integration Points
- **Seller Product Creation** (`seller/views.py:add_product`): Show AI price suggestion before save
- **ProductVariant** (`shop.models.ProductVariant`): Store `ai_suggested_price`, `ai_confidence`, `pricing_job_id`
- **Deals/Flash Sales** (`deals.models.Deal`): Auto-suggest deal prices
- **Seller Dashboard**: Price optimization alerts, margin analysis
- **Blog/Marketing**: Price justification content for storytelling

## Acceptance Criteria
- [ ] Seller uploads product → gets price suggestion in < 5 seconds
- [ ] Recommendation within 15% of actual selling price in 80% of cases (backtest)
- [ ] Floor price never below calculated cost (material + labor + 20% overhead)
- [ ] Confidence score correlates with accuracy (high confidence = high accuracy)
- [ ] Explains reasoning in plain language (Hindi/English)
- [ ] Updates suggestion when raw material costs change significantly
- [ ] Seller feedback (accept/reject/modify) improves future suggestions
- [ ] Works for new/unique products with no direct comparables (uses attribute similarity)

## Dependencies
- #7: AI Infrastructure & Backend Services
- #2: AI Image Enhancer (needs quality images for visual analysis)
- #3: Multilingual Auto-Cataloger (needs structured attributes)
- #5: Unified Seller Input Layer

## Labels: ai, ml, pricing, seller-features, high-priority
